client/src/components/utilities.ts

```diff
@@ -21,7 +21,11 @@ export const api = async <T>(
       // If we get a 401, it most likely means the session expired, so we
       // redirect to the login screen with a flag to show a message.
       if (response.status === 401) {
-        window.location.replace('/#logged-out')
+        window.location.replace(
```

Comment:
I should have been clearer with the guidance I meant to give on this. `window.location.replace` will replace the current history entry with the new one. For popup error messages, ideally you do a `replace` *after* you show the message to remove the error message from the history.. but if you're using a query parameter that's going to reload the page anyways.

So maybe we just live with the error message in the history, and do a `window.location = ` here cause `replace` is almost certainly not what we want.